File: skylotoneoneone_ui/src/services/TagService.js
import http from "../http-common"; 

class TagService {
  getAllTags(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/tag/tags`, searchDTO);
  }

  get(tagId) {
    return this.getRequest(`/tag/${tagId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/tag?field=${matchData}`, null);
  }

  addTag(data) {
    return http.post("/tag/addTag", data);
  }

  update(data) {
  	return http.post("/tag/updateTag", data);
  }
  
  uploadImage(data,tagId) {
  	return http.postForm("/tag/uploadImage/"+tagId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new TagService();

File: src/main/java/com/skylotoneoneone/service/TagService.java
package com.skylotoneoneone.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skylotoneoneone.domain.Tag;
import com.skylotoneoneone.dto.TagDTO;
import com.skylotoneoneone.dto.TagSearchDTO;
import com.skylotoneoneone.dto.TagPageDTO;
import com.skylotoneoneone.dto.TagConvertCriteriaDTO;
import com.skylotoneoneone.service.GenericService;
import com.skylotoneoneone.dto.common.RequestDTO;
import com.skylotoneoneone.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface TagService extends GenericService<Tag, Integer> {

	List<Tag> findAll();

	ResultDTO addTag(TagDTO tagDTO, RequestDTO requestDTO);

	ResultDTO updateTag(TagDTO tagDTO, RequestDTO requestDTO);

    Page<Tag> getAllTags(Pageable pageable);

    Page<Tag> getAllTags(Specification<Tag> spec, Pageable pageable);

	ResponseEntity<TagPageDTO> getTags(TagSearchDTO tagSearchDTO);
	
	List<TagDTO> convertTagsToTagDTOs(List<Tag> tags, TagConvertCriteriaDTO convertCriteria);

	TagDTO getTagDTOById(Integer tagId);







}





